internal/docstore/mongodocstore/codec.go

```diff
@@ -31,8 +31,8 @@ import (
 // This code is copied from memdocstore/codec.go, with some changes:
 // - special treatment for primitive.Binary
 
-func encodeDoc(doc driver.Document) (map[string]interface{}, error) {
-	var e encoder
+func encodeDoc(doc driver.Document, lowercaseFields bool) (map[string]interface{}, error) {
```

Comment:
Consider make the second argument a `encoderOptions` which includes the `lowercaseFields` just like opening a collection.

Reply:
Can we do that if we ever need more than one or two of these? 

We do it for opening a collection to preserve backwards-compatibility. Here we don't have to worry about that, so I'd rather keep things as simple as possible.

Reply:
Yeah that makes sense.